pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/search/ElasticSearch.java

```diff
@@ -185,6 +185,45 @@ public ElasticSearch getParticipantsWithinRange(String esParticipantsIndex, int
         return new ElasticSearch(esParticipants, response.getHits().getTotalHits().value);
     }
 
+    /**
+     * Searches for a participant by shortId in the given ES index.
+     *
+     * @param esParticipantsIndex the ES index to search
+     * @param shortId the shortId to search for
+     * @return an optional containing in form if {@link ElasticSearchParticipantDto} the participant if found, or empty if not found.
+     *     throws a DsmInternalError if more than one participant is found.
+     */
+    public Optional<ElasticSearchParticipantDto> getParticipantByShortId(String esParticipantsIndex, String shortId) {
```

Comment:
I understand what you mean, but the return type is different here (that method returns the map of es data), and we can get rid of the method in the ElasticSearchUtil because that is only used in the Covid19 order registration (which we can clean up too, but it was outside of the scope of this ticket). So I thought not to re-use that because as you pointed out it will be good to move out of ElasticSearchUtil into ElasticSearch and ElasticSearchService.

Reply:
To be clear I suggested that ElasticSearchService was an appropriate place for new code, but did not suggest that ElasticSearch should be expanded. It is sort of a Frankenstein class that we should eventually replace.

In any case, there is no sense in writing new code to do something that existing code already does. The latter has some mileage in production, and the former does not.

I get that the result from `getSingleParticipantFromES` is not directly what you need, but it can be converted. However, there is a ready-made method you can call in ElasticSearchService: getSingleParticipantDocument. It should be straightforward to use and return data in the form you need.